From: csagan5 <32685696+csagan5@users.noreply.github.com>
Date: Fri, 17 Jul 2020 09:28:36 +0200
Subject: Enable legacy TLS interstitital warning

Expose show-legacy-tls-warnings flag on Android as well
---
 chrome/browser/about_flags.cc                 |  2 --
 chrome/browser/ssl/tls_deprecation_config.cc  | 22 +------------------
 components/security_state/core/features.cc    |  2 +-
 net/base/features.cc                          |  2 +-
 .../network/legacy_tls_config_distributor.cc  | 21 +-----------------
 services/network/ssl_config_service_mojo.cc   |  7 ++----
 6 files changed, 6 insertions(+), 50 deletions(-)

diff --git a/chrome/browser/about_flags.cc b/chrome/browser/about_flags.cc
--- a/chrome/browser/about_flags.cc
+++ b/chrome/browser/about_flags.cc
@@ -6045,11 +6045,9 @@ const FeatureEntry kFeatureEntries[] = {
      FEATURE_VALUE_TYPE(features::kElasticOverscrollWin)},
 #endif
 
-#if !defined(OS_ANDROID)
     {"show-legacy-tls-warnings", flag_descriptions::kLegacyTLSWarningsName,
      flag_descriptions::kLegacyTLSWarningsDescription, kOsDesktop,
      FEATURE_VALUE_TYPE(security_state::features::kLegacyTLSWarnings)},
-#endif
 
 #if defined(OS_CHROMEOS)
     {"enable-assistant-aec", flag_descriptions::kEnableGoogleAssistantAecName,
diff --git a/chrome/browser/ssl/tls_deprecation_config.cc b/chrome/browser/ssl/tls_deprecation_config.cc
--- a/chrome/browser/ssl/tls_deprecation_config.cc
+++ b/chrome/browser/ssl/tls_deprecation_config.cc
@@ -58,27 +58,7 @@ void SetRemoteTLSDeprecationConfig(const std::string& binary_config) {
 }
 
 bool ShouldSuppressLegacyTLSWarning(const GURL& url) {
-  if (!url.has_host() || !url.SchemeIsCryptographic())
-    return false;
-
-  auto* proto = TLSDeprecationConfigSingleton::GetInstance().GetProto();
-  // If the config is not yet loaded, we err on the side of not showing warnings
-  // for any sites.
-  if (!proto)
-    return true;
-
-  // Convert bytes from crypto::SHA256 so we can compare to the proto contents.
-  std::string host_hash_bytes = crypto::SHA256HashString(url.host_piece());
-  std::string host_hash = base::ToLowerASCII(
-      base::HexEncode(host_hash_bytes.c_str(), host_hash_bytes.size()));
-  const auto& control_site_hashes = proto->control_site_hashes();
-
-  // Perform binary search on the sorted list of control site hashes to check
-  // if the input URL's hostname is included.
-  auto lower = std::lower_bound(control_site_hashes.begin(),
-                                control_site_hashes.end(), host_hash);
-
-  return lower != control_site_hashes.end() && *lower == host_hash;
+  return false;
 }
 
 void ResetTLSDeprecationConfigForTesting() {
diff --git a/components/security_state/core/features.cc b/components/security_state/core/features.cc
--- a/components/security_state/core/features.cc
+++ b/components/security_state/core/features.cc
@@ -16,7 +16,7 @@ const char kMarkHttpAsParameterWarningAndDangerousOnFormEdits[] =
 const char kMarkHttpAsParameterDangerWarning[] = "danger-warning";
 
 const base::Feature kLegacyTLSWarnings{"LegacyTLSWarnings",
-                                       base::FEATURE_DISABLED_BY_DEFAULT};
+                                       base::FEATURE_ENABLED_BY_DEFAULT};
 
 const base::Feature kSafetyTipUI{"SafetyTip",
                                  base::FEATURE_DISABLED_BY_DEFAULT};
diff --git a/net/base/features.cc b/net/base/features.cc
--- a/net/base/features.cc
+++ b/net/base/features.cc
@@ -152,7 +152,7 @@ const base::Feature kTurnOffStreamingMediaCachingAlways{
     "TurnOffStreamingMediaCachingAlways", base::FEATURE_DISABLED_BY_DEFAULT};
 
 const base::Feature kLegacyTLSEnforced{"LegacyTLSEnforced",
-                                       base::FEATURE_DISABLED_BY_DEFAULT};
+                                       base::FEATURE_ENABLED_BY_DEFAULT};
 
 const base::Feature kSchemefulSameSite{"SchemefulSameSite",
                                        base::FEATURE_DISABLED_BY_DEFAULT};
diff --git a/services/network/legacy_tls_config_distributor.cc b/services/network/legacy_tls_config_distributor.cc
--- a/services/network/legacy_tls_config_distributor.cc
+++ b/services/network/legacy_tls_config_distributor.cc
@@ -51,26 +51,7 @@ scoped_refptr<LegacyTLSExperimentConfig> LegacyTLSExperimentConfig::Parse(
 
 bool LegacyTLSExperimentConfig::ShouldSuppressLegacyTLSWarning(
     const std::string& hostname) const {
-  // Match on eTLD+1 rather than full hostname (to account for subdomains and
-  // redirects). If no registrable domain is found, default to using the
-  // hostname as-is.
-  auto domain = net::registry_controlled_domains::GetDomainAndRegistry(
-      hostname, net::registry_controlled_domains::INCLUDE_PRIVATE_REGISTRIES);
-  if (domain.empty())
-    domain = hostname;
-
-  // Convert bytes from crypto::SHA256 so we can compare to the proto contents.
-  std::string host_hash_bytes = crypto::SHA256HashString(domain);
-  std::string host_hash = base::ToLowerASCII(
-      base::HexEncode(host_hash_bytes.data(), host_hash_bytes.size()));
-  const auto& control_site_hashes = proto_.control_site_hashes();
-
-  // Perform binary search on the sorted list of control site hashes to check
-  // if the input URL's hostname is included.
-  auto lower = std::lower_bound(control_site_hashes.begin(),
-                                control_site_hashes.end(), host_hash);
-
-  return lower != control_site_hashes.end() && *lower == host_hash;
+  return false;
 }
 
 LegacyTLSConfigDistributor::LegacyTLSConfigDistributor() = default;
diff --git a/services/network/ssl_config_service_mojo.cc b/services/network/ssl_config_service_mojo.cc
--- a/services/network/ssl_config_service_mojo.cc
+++ b/services/network/ssl_config_service_mojo.cc
@@ -120,11 +120,8 @@ bool SSLConfigServiceMojo::CanShareConnectionWithClientCerts(
 
 bool SSLConfigServiceMojo::ShouldSuppressLegacyTLSWarning(
     const std::string& hostname) const {
-  // If the config is not yet loaded, we err on the side of not showing warnings
-  // for any sites.
-  if (!legacy_tls_config_)
-    return true;
-  return legacy_tls_config_->ShouldSuppressLegacyTLSWarning(hostname);
+  // never suppress any warning
+  return false;
 }
 
 void SSLConfigServiceMojo::OnNewCRLSet(scoped_refptr<net::CRLSet> crl_set) {
-- 
2.17.1

